src/connector/src/parser/scalar_adapter.rs

```diff
@@ -315,6 +315,22 @@ impl ScalarAdapter {
     }
 }
 
+pub fn validate_pg_type_to_rw_type(pg_type: &DataType, rw_type: &DataType) -> bool {
+    if pg_type == rw_type {
+        return true;
+    }
+    match rw_type {
+        DataType::Varchar => matches!(pg_type, DataType::Decimal | DataType::Int256),
+        DataType::List(box DataType::Varchar) => {
+            matches!(
+                pg_type,
+                DataType::List(box (DataType::Decimal | DataType::Int256))
+            )
+        }
```

Comment:
Is the same check implemented in the Java version?

Reply:
I don't Java version validates type matching, only column names. I added this extra part. https://github.com/risingwavelabs/risingwave/blob/5d6aaf47710b449cde54081e987ba8d457adb6c6/java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java#L46-L119.

cc @StrikeW could you confirm if my understanding is correct?

Reply:
The data type validation is indeed a missing piece of the JDBC sink.